server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -925,6 +927,30 @@ def duplicate(
         )
         return StepContext(query, columns + [step.new_column_name])
 
+    @classmethod
+    def _interval_to_seconds(cls, value: Selectable) -> functions.Function:
+        raise NotImplementedError(
+            f"[{cls.DIALECT}] Converting an INTERVAL to seconds is not supported"
+        )
+
+    def duration(
+        self: Self,
+        *,
+        builder: "QueryBuilder",
+        prev_step_table: str,
+        columns: list[str],
+        step: "DurationStep",
+    ) -> StepContext:
+        the_table = Table(prev_step_table)
+        as_seconds = self._interval_to_seconds(
+            self._cast_to_timestamp(the_table[step.end_date_column])
+            - self._cast_to_timestamp(the_table[step.start_date_column]),
+        )
+        new_column = (as_seconds / DURATIONS_IN_SECOND[step.duration_in]).as_(step.new_column_name)
```

Comment:
Can the division output decimal values?
Would be nice to add a test case for that (like: an duration of 1 hour and a half should appear as 1.5 hours)

Reply:
You're right, I'll see if I can craft such as fixture

Reply:
[84d9c03](https://github.com/ToucanToco/weaverbird/pull/1878/commits/84d9c037045648df588e14983555a47d2867d937)